### DISCUSSION TITLE
deep filtering nest objects/arrays

### DISCUSSION THREAD

**Discussion by singhjaskaran2002** [directus/directus#18335] [Archive]
### Subject Area

Filter rules

### Suggestions

There is a GET request:

`https://content.gfk.com/items/Config_Contact?filter[translations][languages_code][_eq]=en-GB&deep[translations][_filter][languages_code][_eq]=en-GB&fields=translations.*`

Sample data is:
```
[
	{
		"translations": [
			{
				"id": 2,
				"name": "German",
				"languages_code": "de-AT"
			},
			{
				"id": 3,
				"name": "English",
				"languages_code": "en-GB"
			}
		]
	}
]
```

Not knowing the exact way to write this query in directus SDK query format for `readByQuery` method. I need translations data for single `languages_code` only. Hitting this API through postman working well but not mentioned the way to write this filter rule using **deep** in `@directus/sdk` filter object.

Sample code for query i am using

```
directus.items('collection_name').readByQuery({
    fields: 'translations.*',
    filter: {
      translations: {
        languages_code : 'en-GB'
      }
    },
});
```

But getting data for all languages instead of single language.

**Comment by micheljacquot34** [comment]
Maybe the filter needs the _eq, something like:

    filter: {
      translations: {
        languages_code : {
			'_eq':  'en-GB'
        }
      }
    },
